core/coreapi/dag.go

```diff
@@ -0,0 +1,61 @@
+package coreapi
+
+import (
+	"context"
+	"fmt"
+	"io"
+	gopath "path"
+
+	coreiface "github.com/ipfs/go-ipfs/core/coreapi/interface"
+	coredag "github.com/ipfs/go-ipfs/core/coredag"
+
+	cid "gx/ipfs/QmNp85zy9RLrQ5oQD4hPyS39ezrrXpcaa7R4Y9kxdWQLLQ/go-cid"
+)
+
+type DagAPI CoreAPI
+
+func (api *DagAPI) Add(ctx context.Context, src io.Reader, inputEnc string, format cid.Prefix) ([]coreiface.Node, error) {
+	nds, err := coredag.ParseInputs(inputEnc, cid.CodecToStr[format.Codec], src, format.MhType, format.MhLength)
```

Comment:
we should probably validate format.Codec, otherwise we would return bad data from the `cid.CodeToStr` map